agent/cache/cache.go

```diff
@@ -269,17 +288,18 @@ func entryExceedsMaxAge(maxAge time.Duration, entry cacheEntry) bool {
 // getWithIndex implements the main Get functionality but allows internal
 // callers (Watch) to manipulate the blocking index separately from the actual
 // request object.
-func (c *Cache) getWithIndex(tEntry typeEntry, r Request, minIndex uint64) (interface{}, ResultMeta, error) {
-	info := r.CacheInfo()
+func (c *Cache) getWithIndex(r request) (interface{}, ResultMeta, error) {
+	info := r.Info
```

Comment:
Could we instead of creating a variable just use `r.Info` since it is no longer a function?

Reply:
Ya, definitely! I did it this way for now to keep the diff smaller, but I think it would make sense to remove the variable and use `r.Info` directly. I'll look to make that change before merging.

Using `r.Info` directly is much safer I think,  in case we modify `info` in this function. It could be possible that we forget to set `r.Info = info` before `r` is passed to another function.